aurelia-slickgrid/src/aurelia-slickgrid/editors/singleSelectEditor.ts

```diff
@@ -0,0 +1,176 @@
+   */
+  columnDef: Column;
+  /**
+   * The multiple-select options for a single select
+   */
+  defaultOptions: any;
+  /**
+   * The default item value that is set
+   */
+  defaultValue: any;
+  /**
+   * The options label/value object to use in the select list
+   */
+  optionCollection: SelectOption[] = [];
+  /**
+   * The property name for values in the optionCollection
+   */
+  valueName: string;
+  /**
+   * The property name for labels in the optionCollection
+   */
+  labelName: string;
+
+  constructor(private args: any) {
+    this.defaultOptions = {
+      container: 'body',
+      filter: false,
+      maxHeight: 200,
+      width: '100%',
+      single: true
+    };
+
+    this.init();
+  }
+
+  /**
+   * The current selected value from the optionCollection
+   */
+  get currentValue() {
+    return this.optionCollection.findOrDefault(c =>
+      c[this.valueName].toString() === this.$filterElm.val())[this.valueName];
+  }
+
+  init() {
+    if (!this.args) {
+      throw new Error('[Aurelia-SlickGrid] A filter must always have an "init()" ' +
+        'with valid arguments.');
+    }
+
+    this.columnDef = this.args.column;
+
+    const filterTemplate = this.buildTemplateHtmlString();
+
+    this.createDomElement(filterTemplate);
+  }
+
+  applyValue(item: any, state: any): void {
+    item[this.args.column.field] = state;
+  }
+
+  destroy() {
+    this.$filterElm.remove();
+  }
+
+  loadValue(item: any): void {
+    // convert to string because that is how the DOM will return these values
+    this.defaultValue = item[this.columnDef.field].toString();
+
+    this.$filterElm.find('option').each((i: number, $e: any) => {
+      if (this.defaultValue.indexOf($e.value) !== -1) {
+        $e.selected = true;
+      } else {
+        $e.selected = false;
+      }
+    });
+
+    this.refresh();
+  }
+
+  serializeValue(): any {
+    return this.currentValue;
+  }
+
+  focus() {
+    this.$filterElm.focus();
+  }
+
+  isValueChanged(): boolean {
+    return this.$filterElm.val() !== this.defaultValue;
+  }
+
+  validate() {
+    if (this.args.column.validator) {
+      const validationResults = this.args.column.validator(this.currentValue, this.args);
+      if (!validationResults.valid) {
+        return validationResults;
+      }
+    }
+
+    return {
+      valid: true,
+      msg: null
+    };
+  }
+
+  private buildTemplateHtmlString() {
+    if (!this.columnDef || !this.columnDef.filter || !this.columnDef.filter.collection) {
+      throw new Error('[Aurelia-SlickGrid] You need to pass a "collection" for ' +
+        'the MultipleSelect Filter to work correctly. Also each option should include ' +
+        'a value/label pair (or value/labelKey when using Locale). For example:: ' +
+        '{ filter: type: FilterType.multipleSelect, collection: [{ value: true, label: \'True\' }, ' +
+        '{ value: false, label: \'False\'}] }');
+    }
+    this.optionCollection = this.columnDef.filter.collection || [];
```

Comment:
I can change it to reference `params.collection`. That should be no problem.

So do you want me to change the `optionCollection` property name in the editor to be named `collection`? `optionCollection` should be a private property in the editor since it is just a reference to `collection`, and to be honest I cannot remember why I called the property `optionCollection`

Reply:
What about accessing the `customStructure`? I use this in both the editors and the `collectionFormatter`? Should that be `params.customStructure` too?

Reply:
And the reason I chose to use the `filter` object is because i assume that most of the time the `filter.collection` and `filter.customStructure` would be the same for the editors `collection` and `customStructure`

Reply:
To make it to work in both Editor and Filter, we will have to review all the Filters, for example the MultipleFilter should use this instead 
```diff
-if (!this.columnDef || !this.columnDef.filter || !this.columnDef.filter.collection) {
+if (!this.columnDef || !this.columnDef.filter || !this.columnDef.filter.collection || this.columnDef.params.collection) {
  throw new Error(`[Aurelia-SlickGrid] You need to pass a "collection" for the MultipleSelect Filter to work correctly. Also each option should include a value/label pair (or value/labelKey when using Locale). For example:: { filter: type: FilterType.multipleSelect, collection: [{ value: true, label: 'True' }, { value: false, label: 'False'}] }`);
}

-const optionCollection = this.columnDef.filter.collection || [];
+const optionCollection = this.columnDef.filter.collection || this.columnDef.params.collection || [];
```

You can keep the `optionCollection` within the editor if you want but the one in the interface should really be `collection` to align with the already implemented Filters. As mentioned in my previous post, we can revisit that in `2.x` 

I didn't think about the `customStructure` but yeah it also makes sense to put it in the `params` as well. I understand the `filter` principle but still I would rather make a perfect cut between the 2 and not see anything about `filter` within an `editor`, it might confuse some users to do that, however `params` is generic enough not to confuse them.

Reply:
`params` should be fine, the only time you might run into problems if there is more than one `collection` or `customStructure`. Not sure if that will ever happen in one column though?

Okay i see where i took `optionCollection` from now. You use it as a private `const` and I just promoted it to a property on the `this` variable within the editor so I could use it in other methods within the class. I really should make all those variables that are not part of the `Editor` interface private, for example
```typescript
  /**
   * The options label/value object to use in the select list
   */
  private _collection: SelectOption[] = [];
  /**
   * The property name for values in the optionCollection
   */
  private _valueName: string;
  /**
   * The property name for labels in the optionCollection
   */
  private _labelName: string;
```

Reply:
oh ok, if you think it should be changed in Filter as well, I'm ok with that. Thanks for the comment

Reply:
we don't have to do that right now because it might be better to change them all later in one commit **if** you want them all explicitly private. I will just change the `optionCollection` name

Reply:
sure, that is more then ok too.

Reply:
oh I guess that would be another thing that we can put inside the params and make the Editor and Filter share it, when provided

Reply:
i can see `elementOptions` because it is for the actual dom element, but do we need to specify it is for the collection element? `collectionOptions`?

Reply:
I think it's more for the `multiple-select` DOM element, so I would go more with `elementOptions` or `domElementOptions` but it's prob too long

Reply:
I can go with `elementOptions`. The only downside I can see with this approach is if there were ever other elements in the column, then they would potentially have to share this `elementOptions`, but this might be how you want it designed since most DOM elements wont share many properties.

It will be a bit before I update this PR, but I will have it to you later this evening.